### PR TITLE
Eq5 155 superviseur voir etat stage student

### PR DESCRIPTION
**Pull request by Ishnewalon**

### CODE REVIEW

File: backend/src/test/java/com/gestionnaire_de_stage/service/SupervisorServiceTest.java [hunk 1]
```diff
@@ -28,6 +28,9 @@
     @Mock
     SupervisorRepository supervisorRepository;
```
Comment (OussaKab) [inline backend/src/test/java/com/gestionnaire_de_stage/service/SupervisorServiceTest.java:29]: add private pls

File: backend/src/test/java/com/gestionnaire_de_stage/service/SupervisorServiceTest.java [hunk 2]
```diff
@@ -28,6 +28,9 @@
...
 
+    @Mock
+    OfferApplicationService offerApplicationService;
```
Comment (OussaKab) [inline backend/src/test/java/com/gestionnaire_de_stage/service/SupervisorServiceTest.java:32]: add private pls

File: backend/src/main/java/com/gestionnaire_de_stage/controller/SupervisorController.java [hunk 3]
```diff
@@ -85,4 +86,24 @@ public SupervisorController(SupervisorService supervisorService, StudentService
         String response = assign ? "Assignement fait!" : "Assignement rejeté, l'étudiant est déjà assigné!";
         return ResponseEntity.ok(new ResponseMessage(response));
     }
+
+    @GetMapping("/studentsStatus/{supervisor_id}")
```
Comment (OussaKab) [inline backend/src/main/java/com/gestionnaire_de_stage/controller/SupervisorController.java:90]: underscore or status pls no camel case for links

**Review by BertSa — APPROVED** [review]
Guddo jobu

File: gestionnaire_de_stage_frontend/src/services/user-service.js [hunk 4]
```diff
@@ -19,3 +19,8 @@ export async function assignStudentToSupervisor(idStudent, idSupervisor) {//TODO
         requestInit(methods.POST, obj));
     return await response.json();
 }
+
+export async function getStudentsStatus(idSupervisor) {
+    const response = await fetch(`${urlBackend}/supervisor/studentsStatus/${idSupervisor}`, requestInit(methods.GET));
```
Comment (OussaKab) [inline gestionnaire_de_stage_frontend/src/services/user-service.js:24]: underscore or status pls no camel case for links

**Review by OussaKab — CHANGES_REQUESTED** [review]

File: backend/src/main/java/com/gestionnaire_de_stage/service/OfferApplicationService.java [hunk 5]
```diff
@@ -124,4 +124,8 @@ public String updateStatus(UpdateStatusDTO updateStatusDTO) throws IdDoesNotExis
         offerApplicationRepository.save(offerApplication);
         return updateStatusDTO.isAccepted()?"Status changé, attendez la signature du contrat" : "Status changé, stage refusé";
     }
+
+    public List<OfferApplication> getAllBySupervisorId(Long supervisor_id) {
```
Comment (FrostDistrict) [inline backend/src/main/java/com/gestionnaire_de_stage/service/OfferApplicationService.java:128]: un checked param?????

File: backend/src/main/java/com/gestionnaire_de_stage/service/SupervisorService.java [hunk 6]
```diff
@@ -70,6 +75,11 @@ public Supervisor getOneByEmailAndPassword(String email, String password) throws
         return supervisorRepository.findSupervisorByEmailAndPassword(email, password);
     }
 
+    public List<OfferApplication> getStudentsStatus(Supervisor supervisor) {
+        List<OfferApplication> offerApplicationList = offerApplicationService.getAllBySupervisorId(supervisor.getId());
```
Comment (FrostDistrict) [inline backend/src/main/java/com/gestionnaire_de_stage/service/SupervisorService.java:79]: unchecked params?????

**Review by OussaKab — APPROVED** [review]
Nice!

**Review by FrostDistrict — CHANGES_REQUESTED** [review]

**Review by FrostDistrict — APPROVED** [review]

**Pull request merged**